docs/source/developer_guide/metadata.md

```diff
@@ -148,5 +158,69 @@ tags = runtime_configuration.metadata['tags']
 
 ```
 
+## Bring Your Own Schemas
+The metadata service enables the ability to introduce your own schemaspaces and bring custom schemas to existing schemaspaces via [entrypoints](https://entrypoints.readthedocs.io/en/latest/).  It is necessary to introduce Schemaspaces when developing a new extension or adding functionality in which it makes sense to group instances of the functionality together.  For example, Code Snippets consist of their own Schemaspace and, currently, are based on a single schema.
+Applications extending _existing_ functionality do not need to introduce a new Schemaspace, since one will already exist, but must be able to introduce their custom schema.  [Introducing your own _runtime_](../developer_guide/pipelines.html#pipeline-processor-customization) would be an example where a Schemaspace is not required.  In this case, only a _`SchemasProvider`_ would be necessary.
+
+### Schemaspace
+A Schemaspace instance is used to represent a set of schemas.  Custom schemaspace classes must derive from the `Schemaspace` base class and are responsible for providing basic information like `id` (UUID), `name` (used as a symbolic name for lookups), `display_name` (a human-readible, descriptive name) and `description`.  They also contain a dictionary of JSON schemas, indexed by schema name.  However, the dictionary is empty at construction and filled during system initialization via the `SchemasProvider` mechanism.  By convention, schemaspace names are plural, while schema names are singular.
+
+#### Registration
+Schemaspaces are discovered via the entrypoints functionality.  When the `SchemaManager` starts, it fetches all entrypoint instances named by the group `'metadata.schemaspaces'`, and uses the entrypoint data to load each found instance.
+Entrypoints are registered in the package's setup configuration.  
+
+Here is an example of the schemaspace entrypoints defined in Elyra:
+```
+entry_points={
+  'metadata.schemaspaces': [
+    'runtimes = elyra.metadata.schemaspaces:Runtimes',
+    'runtimes-images = elyra.metadata.schemaspaces:RuntimeImages',
+    'code-snippets = elyra.metadata.schemaspaces:CodeSnippets',
+    'component-registries = elyra.metadata.schemaspaces:ComponentRegistries'
+  ]
+}
+```
+As you can see, each entry has a _name_ that equates to a module and object/class.
+
+### SchemasProvider
+As mentioned above, `Schemaspace` instances do not know their set of schemas at the time of their construction.  Instead, the `SchemaManager` loads a different entrypoints group following the loading of schemaspaces named `'metadata.schemas_providers'`.  These entrypoints identify a `SchemasProvider` object and this object is responsible for returning schemas when called upon by the `SchemaManager`.
+`SchemasProvider` instances are concrete instances of the abstract base class `SchemasProvider`.  The class is really quite simple:
```

Comment:
Can you reword this (`really quite simple`) please so this does not come across as judgemental?